docs/rep/controller/openapi.yaml

```diff
@@ -0,0 +1,298 @@
+openapi: 3.0.2
+info:
+  title: Rik
+  version: 0.1.0
+paths:
+  /api/v0/workloads.list:
+    get:
+      tags:
+        - Workloads
+      description: List all workload
+      parameters:
+        - required: false
+          schema:
+            title: limit
+            minimum: 1
+            type: integer
+          name: limit
+          in: query
+      responses:
+        '200':
+          description: OK
+          content:
+            text/plain:
+              schema:
+                 $ref: '#/components/schemas/WorkloadList'
+  /api/v0/workloads.create:
+    post:
+      tags:
+        - Workloads
+      description: Create a new workload
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/WorkloadCreate'
+      responses:
+        '200':
+          description: Successful Response
+  /api/v0/workloads.delete:
+    post:
+      tags:
+        - Workloads
+      description: Delete a workload
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/WorkloadId'
```

Comment:
A workload, tenant or instance id is just a string, and it should be a string representation of a 128 bit [UUID](https://commons.apache.org/sandbox/commons-id/uuid.html). There's no need for a special schema for it, the UUID uniquely identifies any item in the cluster.

Reply:
We also want to delete a workload or instance by its name like we can with docker or k8s.

Reply:
But for tenants and instances I will change it so we only keep id.

Reply:
> We also want to delete a workload or instance by its name like we do with docker or k8s.

You can implement that with 2 API calls (One to get the list of instances/workloads, then get the uuid for the name you're looking for, and then another one for deleting with the uuid you resolved). Keeping your API simple and consistent, and flexible enough to build easy to use UX on top of it is what matters the most.